lib/MusicBrainz/Server/WebService/JSONSerializer.pm

```diff
@@ -14,53 +15,33 @@ sub fmt { 'json' }
 
 sub serialize
 {
-    my ($self, $type, @data) = @_;
+    my ($self, $type, $entity, $inc, $opts) = @_;
 
     $type =~ tr/-/_/;
 
-    my $override = $self->meta->find_method_by_name($type);
-    return $override->execute($self, @data) if $override;
+    if ($type =~ m/^(.*)_list$/ && exists($entity->{items})) {
+      if (exists($ENTITIES{$1}) && exists($ENTITIES{$1}->{plural_url})) {
+        $type = $1;
+        my $singular = $ENTITIES{$type}->{url};
+        my $plural   = $ENTITIES{$type}->{plural_url};
+        my $ret = {
+          $plural => [ map { serialize_entity($_, $inc, $opts, 1) } sort_by { $_->gid } @{ $entity->{items} } ],
+        };
+        if (defined($entity->{offset}) || defined($entity->{total})) {
+          $ret->{$singular . '-offset'} = number($entity->{offset});
+          $ret->{$singular . '-count' } = number($entity->{total});
+        }
+        return encode_json($ret);
+      }
+    }
 
-    my ($entity, $inc, $opts) = @data;
+    my $override = $self->meta->find_method_by_name($type);
+    return $override->execute($self, $entity, $inc, $opts) if $override;
```

Comment:
Moving this here means that it is impossible to override, e.g., `label_list` or any of the lists, should that become necessary. So I would prefer to keep the override first.